common/source/docs/common-fettec-onewire.rst

```diff
@@ -0,0 +1,146 @@
+Ardupilot needs the full-duplex implementation of OneWire. Connect the telemetry output from the ESC to the serial RX of your flight-controller.
+The signal-pins from the ESC are used to receive the commands from the flight-controller, so the serial TX from the flight-controller must be connected to 
+every signal-input in parallel. On Pixhawk Cube Black telem1 port is tested successfully and recommended.
+
+
+Configuring
+===========
+Presetup: 
+Each ESC has it's own ID, which can be configured by the ESC Configuration tool. 
+Duplicate IDs are invalid.
+If two or more 4in1 ESCs should be connected, the IDs of the second and third ESC have to be changed before they are connected.
+
+The ESC IDs have to start from ID 1 and increase without a gap.
+
+If there should be additional ESCs or Servos with different protocol (like PWM), they have to be added at the end. The servo number corresponds directly to the OneWire ESC ID. 
+E.g.:
+Servo1 is connected to OneWire ID 1
+Servo2 is connected to OneWire ID 2
+Servo3 is connected to OneWire ID 3
+Servo4 is connected to OneWire ID 4
+Servo5 and following can then be connected to a PWM channel
+The function of the servo can be set like described below.
+
+The following parameters have to be changed in MissionPlanner under *CONFIG* -> *Full Parameter List* or with other Mavlink capable ground control station software.
+
+Set the SERIALx_PROTOCOL parameter to ``FETtec_OneWire`` (38) and SERIALx_OPTIONS parameter to ``HalfDuplex`` (4), where x is the number of the serial port you are using.
+
+For example, a connection to SERIAL1 port would need:
+
+- :ref:`SERIAL1_PROTOCOL<SERIAL5_PROTOCOL>` = 38 (FETtec_OneWire)
+- :ref:`SERIAL1_OPTIONS<SERIAL5_OPTIONS>` = 0 (default, full duplex)
+
+
+This digital protocol can use the full 1000 to 2000 range. Do set these parameters as well:
+
+- :ref:`MOT_PWM_MAX<MOT_PWM_MAX>` = 2000
+- :ref:`MOT_PWM_MIN<MOT_PWM_MIN>` = 1000
+
+
+Change SERVOx_FUNCTION parameters according to the number of motors and their order. e.g. for a quadcopter:
+
+- :ref:`SERVO1_FUNCTION<SERVO1_FUNCTION>` = 33
+- :ref:`SERVO2_FUNCTION<SERVO2_FUNCTION>` = 34
+- :ref:`SERVO3_FUNCTION<SERVO3_FUNCTION>` = 35
+- :ref:`SERVO4_FUNCTION<SERVO4_FUNCTION>` = 36
+
+FTW Mask:
+After that you must set the servo output channels you want to use in the :ref:`SERVO_FTW_MASK<SERVO_FTW_MASK>` parameter, and again restart your autopilot. For example, a quadcopter using the first four motors would need the first four bits of the bitmask set:
+
+:ref:`SERVO_FTW_MASK<SERVO_FTW_MASK>` = 15
+
+Each bit of the mask describes whether a motor is attached and should be used with oneWire. The value is configured decimal and is the sum of the decimal equivalent of each set bit. Bit 1 corresponds to 2^1 = 1, Bit 2 corresponds to 2^2 = 2,... . E.g. if you have connected Motor 1,2,3,4 the mask is 1+2+4+8 = 15.  
+
+Simply sum the decimal values for the motors connected or read it from the list below.
+
++---+----+
```

Comment:
malformed table

Reply:
must make all rows the same form and width as the Motor Count | Mask row...all "|" and "+" must be aligned or the table will not display